paywall/src/components/Paywall.js

```diff
@@ -110,12 +121,29 @@ export const mapStateToProps = ({ locks, keys, modals, router, account }) => {
     }
   })
 
+  const lock = locksFromUri.length ? locksFromUri[0] : { address: 0 }
```

Comment:
`{ address: 0 }` is that supposed to a no-op lock?
Not feeling very good about this. can you explain what is going on?

Reply:
yes. Basically, as I wrote out an explanation, I figured out a simpler way to do this. I'll send it up and request another look.